### DISCUSSION TITLE
Adjusting FPS ports default settings (Doom, Wolf3D, Quake, Duke3D, etc.)

### DISCUSSION THREAD

**Discussion by brunocar** [OnionUI/Onion#648] [Ideas]
Hi Onion devs, in the spirit of the new year i decided to play doom as is my tradition, but this time on the miyoo mini, and for good measure i tried the other FPS ports, i bring some feedback:
Doom/Doom2: by default the rumble is on and is VERY strong, I only found a way to disable it, not reduce it, its on the core options, nothing in game, so while the option is nice and works, I'd suggest having it off by default as it can both be annoying when playing AND be a battery drain. 
As for the controls, the default scheme is good since it mirrors the PSX control scheme, however, I ran into 2 problems trying to change it, first, changing them from the retroarch quick menu, despite listing the functions in game, it seems to emulate keys, which can be confusing to people if they wanna say, change the strafe key and find themselves without an enter key, by changing them in game this problem is fixed, so its more so a UX thing than a bug, and more of an issue with the port than the config.
The other issue is that I flat out cannot get the 180° turn command to work, no matter what I try, it just does nothing, but again, issue with the port.
With all that said, if the 180° turn command worked, i'd suggest putting it where strafe is by default currently, as that command is redundant while the 180° turn is more useful.
Duke 3D: the default controls could use some work, using the menu button as start is rather confusing, since in the build engine thats not just the pause button, but also the accept key in menus. Besides that, shooting being on A and strafing being in R2/L2 is not ergonomic or consistent with the other ports, the other keys could be rearranged with that in mind.
Quake: the default controls are fantastic, tho I would make the default sensitivity a bit lower, but thats personal preference, the one thing I'd say is that using R2/L2 AND X for looking up is a bit extra, i suggest changing X to reset view, to compliment R2/L"
Wolf 3D/ECwolf: Same thing as Duke3D, A for shooting is very unergonomic, specially when the game uses so few keys by default, the rest is mostly fine.
I would also like to add that i've found 2 bugs on this port that could be related to onion's fork of RA: the reset button does not work, the text string shows up but it doesnt reset the game, so the only way of going back to the main menu is dying, and getting hit causes crackling in the sound.

**Comment by schmurtzm** [comment]
Hi @axcelon  thanks for this interesting feedback 🙂
As Aemiii91 said we are working on ports for Onion 4.1 , all the quoted ports will be included.

----

About Doom : Yes the rumble is too strong,  it can varies a little between 2 version of Miyoo .  Changing input -> haptic feedback/vibration should be enough (thanks Totofaki for the advice).

Fun fact rumble is not supported in the libretro documentation : https://docs.libretro.com/library/prboom/ 

The 180° turn command works but it is mapped to R3 which is power button 😉 So indeed it requires to be remapped to be used.

For the iwad names you're right the naming is just a little mess. The core is not well done for that : for example if you put the doom2 wad file in the same folder of SIGIL it will not work as it should so it is impossible to put all the .wad in one folder (so you have to duplicate doom.wad for example) .

----

About Duke, do you think possible to make a better key mapping and then export eduke32.cfg and settings.cfg files ?

----

Wolf 3D/ECwolf : I think these remarks should be addressed in an issue on the official repo
